lte/gateway/c/oai/tasks/nas/nas_state_converter.cpp

```diff
@@ -1514,6 +1515,34 @@ void NasStateConverter::proto_to_emm_security_context(
       emm_security_context_proto.next_hop_chaining_count();
 }
 
+void NasStateConverter::nw_detach_data_to_proto(
+    nw_detach_data_t* detach_timer_arg,
+    oai::NwDetachData* detach_timer_arg_proto) {
+  OAILOG_FUNC_IN(LOG_MME_APP);
+  detach_timer_arg_proto->set_ue_id(detach_timer_arg->ue_id);
+  detach_timer_arg_proto->set_timer_running(true);
```

Comment:
Isn't it sufficient to check that the NwDetachData is present in the EmmContext object?

Reply:
Thank you Shruti for noticing this. Protobuf also provides "has_<field_name>" Using this function, we will get to know whether timer is running or not. So updated the code to use "has_nw_detach_timer_data" to identify that timer is running.